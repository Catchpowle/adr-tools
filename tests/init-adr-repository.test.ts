/* eslint-disable no-sync */
import { describe, it, expect, afterEach } from 'vitest';
import childProcess from 'child_process';
import path from 'path';
import fs from 'fs';

describe('Init an ADR Repository', () => {
  afterEach(() => {
    childProcess.execSync('rm -rf .adr-dir doc tmp', { cwd: __dirname });
  });

  const adr: string = path.resolve(__dirname, '../src/index.ts');
  const command = `npx ts-node ${adr}`;

  it('should use an alternate directory', () => {
    const directory: string = path.resolve(path.join(__dirname, 'doc/adr'));
    childProcess.execSync(`${command} init`, { cwd: __dirname });
    const expectedFile: string = path.join(directory, '0001-record-architecture-decisions.md');
    expect(fs.existsSync(expectedFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
  });

  it('should use an alternate directory', () => {
    const directory: string = path.resolve(path.join(__dirname, 'tmp', 'alternative-dir'));
    childProcess.execSync(`${command} init ${directory}`, { cwd: __dirname });
    childProcess.execSync(`${command} new Example ADR`, { cwd: __dirname });

    const expectedInitFile: string = path.join(directory, '0001-record-architecture-decisions.md');
    expect(fs.existsSync(expectedInitFile)).toBeTruthy();

    const expectedNewFile: string = path.join(directory, '0002-example-adr.md');
    expect(fs.existsSync(expectedNewFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedNewFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
  });
});
