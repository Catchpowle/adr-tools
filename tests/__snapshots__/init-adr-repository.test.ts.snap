// Vitest Snapshot v1

exports[`Init an ADR Repository > should use an alternate directory 1`] = `
"# 1. Record architecture decisions

Date: 2022-06-22

## Status

Accepted

## Context

We need to record the architectural decisions made on this project.

## Decision

We will use Architecture Decision Records, as [described by Michael Nygard](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions).

## Consequences

See Michael Nygard's article, linked above.
For a lightweight ADR toolset, see Nat Pryce's [adr-tools](https://github.com/npryce/adr-tools).
> For a node version of the same tooling, see Meza's [adr-tools](https://github.com/meza/adr-tools).
"
`;

exports[`Init an ADR Repository > should use an alternate directory 2`] = `
"# 2. Example ADR

Date: 2022-06-22

## Status

Accepted

## Context

The issue motivating this decision, and any context that influences or constrains the decision.

## Decision

The change that we're proposing or have agreed to implement.

## Consequences

What becomes easier or more difficult to do and any risks introduced by the change that will need to be mitigated.
"
`;
